api/lib/opentelemetry/trace/tracer.rb

```diff
@@ -25,8 +25,15 @@ class Tracer
       #   span to the block.
       def in_span(name, attributes: nil, links: nil, start_timestamp: nil, kind: nil, with_parent: nil)
         span = nil
-        span = start_span(name, attributes: attributes, links: links, start_timestamp: start_timestamp, kind: kind, with_parent: with_parent)
-        Trace.with_span(span) { |s, c| yield s, c }
+        if with_parent.nil?
+          span = start_span(name, attributes: attributes, links: links, start_timestamp: start_timestamp, kind: kind, with_parent: with_parent)
+          Trace.with_span(span) { |s, c| yield s, c }
+        else
+          OpenTelemetry::Context.with_current(with_parent) do
```

Comment:
Another way to code this would be to make `OpenTelemetry::Context.with_current` yield and return if `with_parent` is nil. That would eliminate the `if` statement in this method.